pyiceberg/table/__init__.py

```diff
@@ -169,15 +171,18 @@ def _check_schema_compatible(table_schema: Schema, other_schema: "pa.Schema") ->
 
     name_mapping = table_schema.name_mapping
     try:
-        task_schema = pyarrow_to_schema(other_schema, name_mapping=name_mapping)
+        other_schema = pyarrow_to_schema(other_schema, name_mapping=name_mapping)
     except ValueError as e:
         other_schema = _pyarrow_to_schema_without_ids(other_schema)
         additional_names = set(other_schema.column_names) - set(table_schema.column_names)
         raise ValueError(
             f"PyArrow table contains more columns: {', '.join(sorted(additional_names))}. Update the schema first (hint, use union_by_name)."
         ) from e
 
-    if table_schema.as_struct() != task_schema.as_struct():
+    missing_table_schema_fields = {field for field in other_schema.fields if field not in table_schema.fields}
```

Comment:
This is a nit - please feel free to take or leave

I found this name confusing because we have `required_table_schema_fields` that are fields from table_schema, and then `missing_table_schema_fields` that are fields from other_schema that are missing in table_schema. `other_schema_fields_missing_in_table_schema` or `fields_missing_in_table_schema` are suggestions that come to my mind that I personally find less confusing.

Reply:
naming is so hard! I had ChatGPT suggest a few candidates. Please LMK if the new fields are still confusing